### ISSUE TITLE
Pistes d'amélioration

### ISSUE THREAD

**Issue by Aurelie-Mutschler** [aurelapsy/stroke#1]
Cellules 11 & 12
C'est une bonne démarche d'essayer de dégager des liens entre l'AVC et les autres features, attention ceci dit car on ne voit pas grand chose en l'état sur les figures, notamment les scatterplots. Pour avoir un meilleur rendu visuel, tu peux essayer de faire des boxplot ou plotter une fonction de régression logistique en utilisant un seul feature et ta variable-cible :
https://app.jedha.co/course/statistics-and-seaborn-ft/linear-relationships-ft
https://app.jedha.co/course/statistics-and-seaborn-ft/categorical-variables-ft

Cellule 20 &21
Très intéressants ces plots 🙂  un boxplot serait le résumé de ces distributions mais on voit bien l'effet des différents features ici. Ce serait bien de mettre ces figures dans ta présentation

Cellule 22
Attention, ici aussi on ne voit pas grand chose sur les scatterplots, à cause de la superposition de points. Il vaudrait mieux faire deux plots différents en séparant les données avec AVC et sans AVC

Cellule 24
Attention, à cause du déséquilibre de classe on ne peut pas vraiment analyser facilement ces barres plot. Il vaudrait mieux faire des analyses en considérants des pourcentages plutôt que des valeurs absolues

Cellules 25 à 29
Pour voir la relation entre le fait de fumer et les AVC, tu pourrais faire un pie chart de la variable Fumeur_ou_non en séparant les data pour AVC = 1 et AVC = 0 (même remarque pour toutes les autres variables catégorielles)

Cellule 30
Je te conseille de séparer dans un notebook différent toute la partie machine learning : c'est plus lisible, mieux du point de vue des performance d'utilisation de ta machine, et ça évite de propager d'éventuelles "erreurs" de manipulation des données dans ton code de machine learning

Cellule 33
- Pourquoi créer des groupes d'âges et ne pas garder la variable âge telle quelle ? (mêmes remarques pour bmi et glucose)
- C'est une bonne idée de faire une imputation différente en tenant compte des groupes d'âge etc, par contre coder l'imputation "à la main" (plutôt que d'utiliser une classe de sklearn) n'est pas une bonne pratique si tu comptes déployer ton modèle (dans une application par exemple)

Cellule 34
J'ai du mal à comprendre ce que tu fais à cette étape ?

Preprocessing 
Attention, il vaut toujours mieux faire un train/test splitting, je n'en vois pas ici ? Tu en auras besoin pour comparer les performances de tes différents modèles

Cellule 36
Pourquoi ne pas utiliser un OneHotEncoder ? Par ailleurs, pour des questions méthodologiques il vaut mieux combiner toutes les étapes de prepreocessing en une seule étape en utilisant les classes Pipeline et ColumnTransformer. Tu peux te référer au template de preprocessing : https://app.jedha.co/course/preprocessing-with-sklearn-course-ft/preprocessing-with-sklearn-ft

Machine Learning
Je te conseille dans un premier temps de ne pas utiliser de technique d'oversampling. C'est une bonne idée mais une fois que tu as fait ça, toute l'évaluation de tes scores va être biaisée. Il est plus prudent de commencer sans et d'évaluer tes scores proprement. 

Je te conseille de te référer aux templates de machine learning fournis en cours et de suivre la démarche générale : train/test splitting, preprocessings sur le train et le test, évaluation des scores sur le train et le test pour diagnostiquer l'overfitting etc... Si tu veux avoir une évaluation de tes métriques qui soit réprésentative de ce qui se passerait dans la pratique clinique et pouvoir conclure sur le meilleur modèle à retenir, il est important d'avoir un test set qui ait les mêmes propriétés que le dataset global et que tu as laissé de côté depuis le départ. Tu pourras ainsi entraîner tes modèles sur le train set et comparer leurs performances sur le test set
